packages/react/src/components/Combobox/ComboboxOption.tsx

```diff
@@ -100,6 +103,19 @@ const ComboboxOption = forwardRef<HTMLLIElement, ComboboxOptionProps>(
       }
     }, [isActive]);
 
+    useEffect(() => {
+      const comboboxValue =
+        typeof propValue !== 'undefined'
+          ? propValue
+          : comboboxOptionRef.current?.innerText;
+
+      if (selectedValue === comboboxValue) {
+        setFormValue(
+          typeof formValue === 'undefined' ? comboboxValue : formValue
+        );
+      }
+    }, [selectedValue, formValue]);
```

Comment:
Should we also watch for changes to `propValue`?
```suggestion
    }, [selectedValue, formValue, propValue]);
```

Although I realize this will only cover when the prop value changes and not the innerText :/

Reply:
Huh, that's an interesting question.

The setting of the form value should really only change when the selected value changes, but I'm now realizing that `formValue` may be unintended. If I select a value, then somewhere in code the value of that field changes I would think the value of the input shouldn't change but let me play around with native inputs to see what the default behavior is in react as we'd likely want to match that.

Reply:
I tested using a native `select` input and updating the values via a `setTimeout` independent of of a user selection.

It does appear that native selects will update the value of the select when an option value changes, even if `<select value={...}>` is set to a no longer valid value, so we probably want to try to mimic that behavior here.

This is probably a bit of an extreme edge case as I'm trying to imagine how often this might pop up. The only scenario where it might be feasible is if the options were returned asynchronously via a fetch request and the same option element had a different value/innerText for the same key. I'm inclined to try to cover what we can here and see if `innerText` does present itself as a problem later on.

Reply:
I think it's also good to keep this in mind: this hook should trigger again upon a selection change, so if a new selection is made the value should re-sync.